framework/mongo/utils.py

```diff
@@ -60,16 +61,78 @@ def wrapper(cls):
         return cls
     return wrapper
 
-
-def get_or_http_error(Model, pk, allow_deleted=False):
-    instance = Model.load(pk)
+def get_or_http_error(Model, pk_or_query, allow_deleted=False, display_name=None):
+    """Load an instance of Model by primary key or modularodm.Q query. Raise an appropriate
+    HTTPError if no record is found or if the query fails to find a unique record
+    :param type Model: StoredObject subclass to query
+    :param pk_or_query:
+    :type pk_or_query: either
+      - a <basestring> representation of the record's primary key, e.g. 'abcdef'
+      - a <QueryBase> subclass query to uniquely select a record, e.g.
+        Q('title', 'eq', 'Entitled') & Q('version', 'eq', 1)
+    :param bool allow_deleted: allow deleleted records?
+    :param basestring display_name:
+    :return: Model instance
+    """
+    display_name = display_name or ''
+
+    instance = None
```

Comment:
This one? It seems to raise an exception- am I missing something? (I'm in 200 if you need to talk over any points in person, incidentally)

```
if not allow_deleted and getattr(instance, 'is_deleted', False):
    raise HTTPError ...
```